<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Описание-Проекта" data-toc-modified-id="Описание-Проекта-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Описание Проекта</a></span><ul class="toc-item"><li><span><a href="#Описание-данных" data-toc-modified-id="Описание-данных-1.1"><span class="toc-item-num">1.1&nbsp;&nbsp;</span>Описание данных</a></span></li><li><span><a href="#Вопросы-на-которые-нужны-ответить-в-процессе-исследования:" data-toc-modified-id="Вопросы-на-которые-нужны-ответить-в-процессе-исследования:-1.2"><span class="toc-item-num">1.2&nbsp;&nbsp;</span>Вопросы на которые нужны ответить в процессе исследования:</a></span></li></ul></li><li><span><a href="#Откроем-таблицу-и-изучим-общую-информацию-о-данных" data-toc-modified-id="Откроем-таблицу-и-изучим-общую-информацию-о-данных-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>Откроем таблицу и изучим общую информацию о данных</a></span></li><li><span><a href="#Предобработка-данных" data-toc-modified-id="Предобработка-данных-3"><span class="toc-item-num">3&nbsp;&nbsp;</span>Предобработка данных</a></span><ul class="toc-item"><li><span><a href="#Удаление-пропусков" data-toc-modified-id="Удаление-пропусков-3.1"><span class="toc-item-num">3.1&nbsp;&nbsp;</span>Удаление пропусков</a></span></li><li><span><a href="#Обработка-аномальных-значений" data-toc-modified-id="Обработка-аномальных-значений-3.2"><span class="toc-item-num">3.2&nbsp;&nbsp;</span>Обработка аномальных значений</a></span></li><li><span><a href="#Изменение-типов-данных" data-toc-modified-id="Изменение-типов-данных-3.3"><span class="toc-item-num">3.3&nbsp;&nbsp;</span>Изменение типов данных</a></span></li><li><span><a href="#Обработка-дубликатов" data-toc-modified-id="Обработка-дубликатов-3.4"><span class="toc-item-num">3.4&nbsp;&nbsp;</span>Обработка дубликатов</a></span></li><li><span><a href="#Категоризация-данных" data-toc-modified-id="Категоризация-данных-3.5"><span class="toc-item-num">3.5&nbsp;&nbsp;</span>Категоризация данных</a></span></li></ul></li><li><span><a href="#Исследование-данных-и-ответ-на-вопросы" data-toc-modified-id="Исследование-данных-и-ответ-на-вопросы-4"><span class="toc-item-num">4&nbsp;&nbsp;</span>Исследование данных и ответ на вопросы</a></span><ul class="toc-item"><li><span><a href="#Есть-ли-зависимость-между-количеством-детей-и-возвратом-кредита-в-срок?" data-toc-modified-id="Есть-ли-зависимость-между-количеством-детей-и-возвратом-кредита-в-срок?-4.1"><span class="toc-item-num">4.1&nbsp;&nbsp;</span>Есть ли зависимость между количеством детей и возвратом кредита в срок?</a></span></li><li><span><a href="#Есть-ли-зависимость-между-семейным-положением-и-возвратом-кредита-в-срок?" data-toc-modified-id="Есть-ли-зависимость-между-семейным-положением-и-возвратом-кредита-в-срок?-4.2"><span class="toc-item-num">4.2&nbsp;&nbsp;</span>Есть ли зависимость между семейным положением и возвратом кредита в срок?</a></span></li><li><span><a href="#Есть-ли-зависимость-между-уровнем-дохода-и-возвратом-кредита-в-срок?" data-toc-modified-id="Есть-ли-зависимость-между-уровнем-дохода-и-возвратом-кредита-в-срок?-4.3"><span class="toc-item-num">4.3&nbsp;&nbsp;</span>Есть ли зависимость между уровнем дохода и возвратом кредита в срок?</a></span></li><li><span><a href="#Как-разные-цели-кредита-влияют-на-его-возврат-в-срок?" data-toc-modified-id="Как-разные-цели-кредита-влияют-на-его-возврат-в-срок?-4.4"><span class="toc-item-num">4.4&nbsp;&nbsp;</span>Как разные цели кредита влияют на его возврат в срок?</a></span></li><li><span><a href="#Возможные-причины-появления-пропусков-в-исходных-данных." data-toc-modified-id="Возможные-причины-появления-пропусков-в-исходных-данных.-4.5"><span class="toc-item-num">4.5&nbsp;&nbsp;</span>Возможные причины появления пропусков в исходных данных.</a></span></li><li><span><a href="#Почему-заполнить-пропуски-медианным-значением-—-лучшее-решение-для-количественных-переменных." data-toc-modified-id="Почему-заполнить-пропуски-медианным-значением-—-лучшее-решение-для-количественных-переменных.-4.6"><span class="toc-item-num">4.6&nbsp;&nbsp;</span>Почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.</a></span></li></ul></li><li><span><a href="#Общий-вывод-исследования:" data-toc-modified-id="Общий-вывод-исследования:-5"><span class="toc-item-num">5&nbsp;&nbsp;</span>Общий вывод исследования:</a></span></li></ul></div>

# Исследование надежности заемщиков

## Описание Проекта

Заказчик — кредитный отдел банка. Нужно разобраться, влияет ли семейное положение и количество детей клиента на факт погашения кредита в срок. Входные данные от банка — статистика о платёжеспособности клиентов.

Результаты исследования будут учтены при построении модели кредитного скоринга — специальной системы, которая оценивает способность потенциального заёмщика вернуть кредит банку.



### Описание данных

- children — количество детей в семье
- days_employed — общий трудовой стаж в днях
- dob_years — возраст клиента в годах
- education — уровень образования клиента
- education_id — идентификатор уровня образования
- family_status — семейное положение
- family_status_id — идентификатор семейного положения
- gender — пол клиента
- income_type — тип занятости
- debt — имел ли задолженность по возврату кредитов
- total_income — ежемесячный доход
- purpose — цель получения кредита

### Вопросы на которые нужны ответить в процессе исследования:

- Есть ли зависимость между количеством детей и возвратом кредита в срок?
- Есть ли зависимость между семейным положением и возвратом кредита в срок?
- Есть ли зависимость между уровнем дохода и возвратом кредита в срок?
- Как разные цели кредита влияют на его возврат в срок?

Так же нужно найти описание для следующих задач:

- Привести возможные причины появления пропусков в исходных данных.
- Объяснить почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

## Откроем таблицу и изучим общую информацию о данных

**Импортируем библиотеку pandas. Прочтем данные из csv-файла в датафрейм и сохраните в переменную `data`.**

In [1]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

In [2]:
# Выведем первые 20 строчек датафрейма `data` на экран.
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Выведем основную информацию о датафрейме с помощью метода `info()`.**

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


**Вывод по первичному анализу данных:**

- В файле 12 колонок, 2 колонки типа float, 5 колонок типа int и 5 колонок типа object.
- В данных обнаружены пропуски, при переработки данных нужно попытаться заполнить их.
- В столбец `total_income` указан неверный формат данных
- В столбце `education` имеются значения передающие один одинаковый смысл, но записаные по-разному. Их нужно привести к общему виду и стандартному регистру

## Предобработка данных

### Удаление пропусков

**Выведем количество пропущенных значений для каждого столбца. Используя комбинацию двух методов.**

In [4]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.

In [5]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработаем значения в этом столбце: заменим все отрицательные значения положительными с помощью метода `abs()`.

In [6]:
data['days_employed'] = data['days_employed'].abs()

**Для каждого типа занятости выведем медианное значение трудового стажа `days_employed` в днях.**

In [7]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставим их как есть. Тем более этот столбец не понадобится нам для исследования.

 **Выведем перечень уникальных значений столбца children.**

In [8]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5], dtype=int64)

В столбце children есть два аномальных значения. Удалим строки, в которых встречаются такие аномальные значения из датафрейма data.

In [9]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

Ещё раз выведем перечень уникальных значений столбца children, чтобы убедиться, что артефакты удалены.

In [10]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5], dtype=int64)

**Заполним пропуски в столбце days_employed медианными значениями по каждого типа занятости income_type.**

In [11]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

Убедимся, что все пропуски заполнены. Проверим себя и ещё раз выведем количество пропущенных значений для каждого столбца с помощью двух методов

In [12]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Заменим вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [13]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Обработаем неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведем их к нижнему регистру. Проверим остальные столбцы.**

In [14]:
data['education'] = data['education'].str.lower()

**Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалим их.**

In [15]:
data.duplicated().sum()

71

In [16]:
data = data.drop_duplicates()

### Категоризация данных

На основании диапазонов, указанных ниже, создадим в датафрейме data столбец total_income_category с категориями:
- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Воспользуемся собственной функцей с именем `categorize_income()` и метод `apply()` для её применения.**

In [17]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [18]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Выведем на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [19]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

Создаим функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Воспользуемся собственной функцией с именем `categorize_purpose()` и метод `apply()`. Изучим данные в столбце `purpose` и определим, какие подстроки помогут нам правильно определить категорию.**

In [20]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [21]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

## Исследование данных и ответ на вопросы

###  Есть ли зависимость между количеством детей и возвратом кредита в срок?

Сгруппируем количество клиентов по количеству детей и вычислим сумму клиентов для каждой группы у которых имелись задолжности по возврату кредитов

In [22]:
grouped_by_children_debt = data.groupby('children').agg({'debt': ['count', 'sum']}) 
grouped_by_children_debt

debt      
          count   sum
children             
0         14091  1063
1          4808   444
2          2052   194
3           330    27
4            41     4
5             9     0

В сгруппированной таблице клиентов по количеству детей вычислим процентное соотноешение клиентов у которых имелись задолжности по возврату кредитов из общего числа.

In [23]:
debt_by_count_children = data.groupby('children')['debt'].sum()
count_of_children = data.groupby('children')['debt'].count()
debt_by_children_procent = (debt_by_count_children/count_of_children)*100
debt_by_children_procent

children
0    7.543822
1    9.234609
2    9.454191
3    8.181818
4    9.756098
5    0.000000
Name: debt, dtype: float64

**Вывод:** В группе клиентов у которых нет детей самый низкий процент(%) задолжностей по возврату кредитов. Но в группах клиентов у которых 4 и 5 детей слишком мало данных чтобы сравнивать все группы между собой, поэтому эти сведения могут быть не объективными.

### Есть ли зависимость между семейным положением и возвратом кредита в срок?

Сгруппируем количество клиентов по семейному статусу и вычислим сумму клиентов для каждой группы у которых имелись задолжности по возврату кредитов

In [24]:
grouped_by_fstatus_debt = data.groupby('family_status').agg({'debt': ['count', 'sum']}) 
grouped_by_fstatus_debt

debt     
                       count  sum
family_status                    
Не женат / не замужем   2796  273
в разводе               1189   84
вдовец / вдова           951   63
гражданский брак        4134  385
женат / замужем        12261  927

В сгруппированной таблице клиентов по семейному статусу вычислим процентное соотноешение клиентов у которых имелись задолжности по возврату кредитов из общего числа.

In [25]:
debt_by_count_fstatus = data.groupby('family_status')['debt'].sum()
count_of_fstatus = data.groupby('family_status')['debt'].count()
debt_by_fstatus_procent = (debt_by_count_fstatus/count_of_fstatus)*100
debt_by_fstatus_procent

family_status
Не женат / не замужем    9.763948
в разводе                7.064760
вдовец / вдова           6.624606
гражданский брак         9.313014
женат / замужем          7.560558
Name: debt, dtype: float64

**Вывод:** Самыми надежными плательщиками оказались: Клиенты со статусом (вдовец / вдова), но также эта группа самая малочисленная и информация может быть не совсем корректной из за малого количества статистики для исследования. Самые ненадежные клиенты со статусом - (Не женат / не замужем) и (гражданский брак) : в этих группах уже больше статистических данных и эту информацию можно учесть при выдачи кридитов.

### Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

Сгруппируем количество клиентов по уровню дохода и вычислим сумму клиентов для каждой группы у которых имелись задолжности по возврату кредитов
Группы клиентов разделены на категории, которые мы обозначили ранее:
Доход 0–30000р — E
Доход 30001–50000р — D
Доход 50001–200000р — C
Доход 200001–1000000р — B
Доход 1000001р и выше — A

In [26]:
grouped_by_income_debt = data.groupby('total_income_category').agg({'debt': ['count', 'sum']}) 
grouped_by_income_debt

debt      
                       count   sum
total_income_category             
A                         25     2
B                       5014   354
C                      15921  1353
D                        349    21
E                         22     2

В сгруппированной таблице клиентов уровню дохода вычислим процентное соотноешение клиентов у которых имелись задолжности по возврату кредитов из общего числа.

In [27]:
debt_by_count_income = data.groupby('total_income_category')['debt'].sum()
count_of_income = data.groupby('total_income_category')['debt'].count()
debt_by_income_procent = (debt_by_count_income/count_of_income)*100
debt_by_income_procent

total_income_category
A    8.000000
B    7.060231
C    8.498210
D    6.017192
E    9.090909
Name: debt, dtype: float64

**Вывод:** Группы клиентов B(Доход 200001–1000000р) и C (Доход 50001–200000р) значительно привосходят по количествую остальные группы, этот метод не подходит для сравнения.

### Как разные цели кредита влияют на его возврат в срок?

Сгруппируем количество клиентов по цели кредита и вычислим сумму клиентов для каждой группы у которых имелись задолжности по возврату кредитов

In [28]:
grouped_by_purpose_debt = data.groupby('purpose_category').agg({'debt': ['count', 'sum']}) 
grouped_by_purpose_debt

debt     
                          count  sum
purpose_category                    
операции с автомобилем     4279  400
операции с недвижимостью  10751  780
получение образования      3988  369
проведение свадьбы         2313  183

В сгруппированной таблице клиентов по цели кредита вычислим процентное соотноешение клиентов у которых имелись задолжности по возврату кредитов из общего числа.

In [29]:
debt_by_count_purpose = data.groupby('purpose_category')['debt'].sum()
count_of_purpose = data.groupby('purpose_category')['debt'].count()
debt_by_income_purpose = (debt_by_count_purpose/count_of_purpose)*100
debt_by_income_purpose

purpose_category
операции с автомобилем      9.347978
операции с недвижимостью    7.255139
получение образования       9.252758
проведение свадьбы          7.911803
Name: debt, dtype: float64

**Вывод:** В этой группе самыми ненадежными заемщиками оказались клиенты которые взяли кредит на получения образования и покупку автомобиля. Самые надежные замещики получили кредит на покупку недвижимости.

### Возможные причины появления пропусков в исходных данных.

**Ответ:** Количество пропусков в столбце days_imployed и total_income одинаковое и пропуски допущены в одних и тех же строках. В этих строках указан общий стаж в днях и ежемесячный доход. Связь отсутвия данных о стаже и доходе может говорит о том что эта информация была не заполнена умышленно и данный клиент не имеет трудового стажа и дохода на данный момент

### Почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

**Ответ:** В данных где есть явные большие группы которые выбиваются из основной статистики может произойти сильное смещение вверх или вниз если использовать средние значения. Поэтому оценивать такие переменные лучше медианой.

##  Общий вывод исследования:

Провели исследование в котором определяли вероятность погашения кредита в срок в зависимости от разных факторов.
    
   Были выведены несколько гипотиз для исследования:
    
    1.Есть ли зависимость между количеством детей и возвратом кредита в срок?
    2.Есть ли зависимость между семейным положением и возвратом кредита в срок?
    3.Есть ли зависимость между уровнем дохода и возвратом кредита в срок?
    4.Как разные цели кредита влияют на его возврат в срок?
    
   В результате исследования выявили следующее:
    
 **Гипотеза №1 подтверждена частично:**
    
     Доля просреченых кредитов среди клиентов без детей -  7.54%
     Доля просреченых кредитов среди клиентов c одним ребенком -  9.23%
     Доля просреченых кредитов среди клиентов c двумя дентьми -  9.45%
   
   Итог по исследуемой зависимости:
   
   Заемщики без детей вероятнее всего вернут кредит, клиенты с одним и двумя детьми возвращат кредиты реже.
   По результатам для клиентов с тремя и более детьми объективное исследование провести нельзя - нужно провести повторное исследование на большей выборке данных
 
 **Гипотеза №2 подтверждена:**
    
     Доля просроченных кредитов среди клиентов в браке -  7.56%
     Доля просроченных кредитов среди клиентов в гражданском браке -  9.31%
     Доля просроченных кредитов среди (не женатых/не замужних) клиентов -  9.45%
     Доля просроченных кредитов среди клиентов в разводе - 7.06%
     Доля просроченных кредитов среди (вдовец/вдова) клиентов - 6.62%
  
   Итог по исследуемой зависимости:
    
   Заемщики в браке\в разводе\ в статусе (вдова\вдовец) вероятнее всего возвращают кредиты в срок по сравнению клиентами находящимися вне брака или в гражданском браке.  Клиентов в разводе\ в статусе (вдова\вдовец) теоретически можно объединять в одну группу при исследовании зависимостей, т.к они так же были в браке ранее ( но в этих группах мало данных для полноценного исследования, следует провести повтороное исследование на большей выборке для этих групп)
    
    
 **Гипотеза №3 подтверждена частично:**
    
     Доля просроченных кредитов среди клиентов с доходом 50001–200000р  -  8.49%
     Доля просроченных кредитов среди клиентов с доходом 200001–1000000р -  7.06%
  
   Итог по исследуемой зависимости:
   
   Замемщики по исследуемым группам С( Доход 50001–200000р) менее вероятно возвращают кредит в срок по сравнению с клиентами группы B ( Доход 200001–1000000р). Клиенты с более большим доходом вероятнее всего вернут кредит в срок по договору.
    Для групп клиентов A,D,E необходимо провести повторное исследование с большей выборкой данных для более точных реузльтатов.
    
    
 **Гипотеза №4 подтверждена:**
    
     Доля просроченных кредитов целью которых была "операции с недвижимостью" -  7.25%
     Доля просроченных кредитов целью которых была "операции с автомобилем"  -  9.34%
     Доля просроченных кредитов целью которых была "проведение свадьбы"  -  7.91%
     Доля просроченных кредитов целью которых была "получаение образования"  - 9.25%
    
   Итог по исследуемой зависимости:
   
   Заемщики целью кредита которых была покупка недвижимости\проведение свадьбы оказались более надежными в сревнении с другими клиентами.
    
    
  **Общий итог исследования:**
    
    
   1) Специалистам которые занимаются разработкой модели кредитного скоринга дли банка необходимо использовать результаты второй и четвертой гипотезы.
    
   2) Необходимо собрать больше данных и провести повторное исследование чтобы получить более точные результаты при проверке первой и третьей гипотезы
    